x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_integrations_client.ts

```diff
@@ -28,32 +23,52 @@ export class RuleMigrationsDataIntegrationsClient extends RuleMigrationsDataBase
   }
 
   /** Indexes an array of integrations to be used with ELSER semantic search queries */
-  async create(): Promise<void> {
+  async populate(): Promise<void> {
     const index = await this.getIndexName();
-    await this.esClient
-      .bulk(
-        {
-          refresh: 'wait_for',
-          operations: INTEGRATIONS.flatMap((integration) => [
-            { update: { _index: index, _id: integration.id } },
-            {
-              doc: {
-                title: integration.title,
-                description: integration.description,
-                data_streams: integration.data_streams,
-                elser_embedding: integration.elser_embedding,
-                '@timestamp': new Date().toISOString(),
+    const packages = await this.dependencies.packageService?.asInternalUser.getPackages();
+    if (packages) {
```

Comment:
Do you want me to move this to a separate function?

Reply:
No need, the current way is fine 👌